src/Symfony/Component/TypeInfo/TypeResolver/PhpDocAwareReflectionTypeResolver.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\TypeInfo\TypeResolver;
+
+use PHPStan\PhpDocParser\Ast\PhpDoc\ParamTagValueNode;
+use PHPStan\PhpDocParser\Ast\PhpDoc\ReturnTagValueNode;
+use PHPStan\PhpDocParser\Ast\PhpDoc\VarTagValueNode;
+use PHPStan\PhpDocParser\Lexer\Lexer;
+use PHPStan\PhpDocParser\Parser\ConstExprParser;
+use PHPStan\PhpDocParser\Parser\PhpDocParser;
+use PHPStan\PhpDocParser\Parser\TokenIterator;
+use PHPStan\PhpDocParser\Parser\TypeParser;
+use Symfony\Component\TypeInfo\Exception\UnsupportedException;
+use Symfony\Component\TypeInfo\Type;
+use Symfony\Component\TypeInfo\TypeContext\TypeContext;
+use Symfony\Component\TypeInfo\TypeContext\TypeContextFactory;
+use Symfony\Component\TypeInfo\TypeResolver\TypeResolverInterface;
+
+/**
+ * Resolves type on reflection priorizing PHP documentation.
+ *
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @internal
+ */
+final readonly class PhpDocAwareReflectionTypeResolver implements TypeResolverInterface
+{
+    private PhpDocParser $phpDocParser;
+    private Lexer $lexer;
+
+    public function __construct(
+        private TypeResolverInterface $reflectionTypeResolver,
+        private TypeResolverInterface $stringTypeResolver,
+        private TypeContextFactory $typeContextFactory,
+    ) {
+        $this->phpDocParser = new PhpDocParser(new TypeParser(), new ConstExprParser());
+        $this->lexer = new Lexer();
+    }
```

Comment:
```suggestion
    private PhpDocParser $phpDocParser;

    public function __construct(
        private TypeResolverInterface $reflectionTypeResolver,
        private TypeResolverInterface $stringTypeResolver,
        private TypeContextFactory $typeContextFactory,
        private Lexer $lexer = new Lexer(),
    ) {
        $this->phpDocParser = new PhpDocParser(new TypeParser(), new ConstExprParser());
    }
```
?

Reply:
```suggestion
    public function __construct(
        private TypeResolverInterface $reflectionTypeResolver,
        private TypeResolverInterface $stringTypeResolver,
        private TypeContextFactory $typeContextFactory,
        private Lexer $lexer = new Lexer(),
        private PhpDocParser $phpDocParser = new PhpDocParser(new TypeParser(), new ConstExprParser()),
    ) {
    }
```
^^

Reply:
Didn't know it is working, thought this is only working for simple instantiation without further arguments 🤔

Reply:
I didn't see the `lexer` and the `phpDocParser` as external/configurable properties, that's why I didn't allowed them to be injected.
Do you think that it worth making them injectable?

Reply:
Yes